src/com/android/launcher3/util/MainThreadInitializedObject.java

```diff
@@ -56,8 +58,17 @@ public T get(Context context) {
 
         if (mValue == null) {
             if (Looper.myLooper() == Looper.getMainLooper()) {
+                Context app;
+                try {
+                    app = context.getApplicationContext();
+                } catch (Exception ignored) {
+                    // https://github.com/LawnchairLauncher/lawnchair/issues/3111
+                    app = LawnchairApp.getInstance();
+                }
```

Comment:
This NPE may occur when `LawnchairLauncher` is restarting.

The other workaround is run `MAIN_EXECUTOR.submit(() -> get(context)).get()` again, WDYT?